website/docs/api/Mock.md

```diff
@@ -111,3 +111,55 @@ interface MatchEvent {
 ### `continue`
 
 This event is being emitted when the network response has neither been overwritten nor interrupted, or if response was already sent by another mock. `requestId` is passed in event callback.
+
+### Examples
+
+Getting a number of pending requests:
+
+```js
+let pendingRequests = 0
+const mock = await browser.mock('**')
+mock.on('request', ({request}) => {
+    pendingRequests++
+    console.log(`matched request to ${request.url}, pending ${pendingRequests} requests`)
+})
+mock.on('match', ({url}) => {
+    pendingRequests--
+    console.log(`resolved request to ${url}, pending ${pendingRequests} requests`)
+})
+```
+
+Throwing an error on 404 network fail:
+
+```js
+const mock = await browser.mock('**')
+mock.on('match', ({url, statusCode}) => {
+    if (statusCode === 404) {
+        throw new Error(`request to ${url} failed with "Not Found"`)
```

Comment:
It throws an unhandled rejection, unless you handle it something like that:

```js
const loadPage = (browser, url) => new Promise(async (resolve, reject) => {
    const mock = await browser.mock('**')
    mock.on('match', ({url, statusCode}) => {
        if (statusCode === 404) {
            reject(new Error(`request to ${url} failed with "Not Found"`))
        }
    })
    await browser.url(url)
    resolve()
})

await loadPage(browser, 'some/url');
```

So yes, if you fail tests on unhandled rejections

Reply:
Gotcha, can we make the example more tangible then by wrapping it into a promise, or wait until the page has loaded and validate for none 404 pages?

Reply:
> can we wait until the page has loaded and validate for none 404 pages?

In this example, or in general?

Resulting example will not be easy to understand. In my tests, this `pendingRequests` goes like:

```
1, 2, 1, 0, 1, 2, 3, ..., 7, 6, 5, ..., 1, 0, 1, 2, 1, 0
                                                       ^
                                                  Page loaded
```

So we would have to add some timeouts (playwright, for example, has 500ms `networkidle` timeout), but it would cause a great performance decrease (even +500ms for one test seems massive)

Reply:
> can we make the example more tangible then by wrapping it into a promise

Sure, 1m